scripts/linters/general_purpose_linter_test.py

```diff
@@ -409,3 +421,21 @@ def test_linter_ignore_pragma(self) -> None:
             [VALID_PY_IGNORE_PRAGMA_FILEPATH], FILE_CACHE)
         lint_task_report = linter.check_bad_patterns()
         self.assertFalse(lint_task_report.failed)
+
+    def test_check_bad_patterns_in_excluded_dirs(self) -> None:
+        bad_pattern_regexp = {
+            'regexp': re.compile(r'TODO[^\(]*[^\)][^:]*[^A-Z]+[^\w]*$'),
```

Comment:
Use a simpler regex that very obviously matches the TODO comment. Try to write a minimal test case that is easy to read and understand.

Reply:
Updated the test making it easy to understand.